features/step_definitions/icon_steps.rb

```diff
@@ -14,6 +14,9 @@
 end
 
 ### WHEN
+When /^I attach an icon with the extension "([^"]*)"$/ do |extension|
```

Comment:
Can you use [Cucumber expressions](https://cucumber.io/docs/cucumber/cucumber-expressions/) for the new step?

Reply:
Thanks for the link, I wasn't aware of those!
I've replaced the regex with an expression.